src/irr_v3d.h

```diff
@@ -29,5 +29,24 @@ typedef core::vector3d<s16> v3s16;
 typedef core::vector3d<u16> v3u16;
 typedef core::vector3d<s32> v3s32;
 
+#if __cplusplus >= 201103L
+
+#include <functional>
+
+// A hash class is required for c++11 unordered containers
+namespace std {
+	template<>
+	struct hash<v3s16> {
+		unsigned operator()(const v3s16 &pos) const
+		{
+			return (0x87654321 * pos.X)
+				^ (0x87654321 * pos.Y)
+				^ (0x87654321 * pos.Z);
```

Comment:
This is a very simplistic hash function and will result in quite a lot collisions.